timApp/static/templates/answerBrowser.html

```diff
@@ -1,175 +1,178 @@
+                        (<a i18n-title title="Link to currently selected answer without other document content"
                             target="_blank"
-                            ng-href="{{ $ctrl.getAnswerLink(true) }}">only</a>)&nbsp;
+                            href="{{ getAnswerLink(true) }}">only</a>)&nbsp;
                         </span>
                     </div>
                 </div>
                 </div>
-                <a ng-if="$ctrl.peerReviewEnabled && !$ctrl.isPeerReview"
+                <a i18n-title *ngIf="peerReviewEnabled && !isPeerReview"
                     title="Link to peer review"
-                ng-href="{{ $ctrl.getReviewLink() }}">Review</a>
+                href="{{ getReviewLink() }}">Review</a>
                 </span>
-             <div ng-show="$ctrl.answers.length == 0 && $ctrl.viewctrl.teacherMode">
-                <span ng-if="!$ctrl.hasUserChanged()">(no answers from the selected user)</span>
-                 <span ng-if="$ctrl.hasUserChanged()"> <tim-loading></tim-loading> </span>
+             <div *ngIf="answers.length == 0 && viewctrl.teacherMode">
+                <span i18n *ngIf="!hasUserChanged()">(no answers from the selected user)</span>
+                 <span *ngIf="hasUserChanged()"> <tim-loading></tim-loading> </span>
              </div>
                  <div class="flex ab-option-row">
-                            <form ng-submit="$ctrl.savePoints()" class="point-form form-inline">
-                                <label class="inline" ng-show="$ctrl.showTeacher() || ($ctrl.giveCustomPoints && $ctrl.allowCustomPoints())">
-                                    Points:<input class="form-control input-xs"
-                                                  ng-class="{ unsaved: $ctrl.points != $ctrl.selectedAnswer.points,
-                                                  'no-step': !$ctrl.markupSettings.pointsStep }"
-                                                  focus-me="$ctrl.shouldFocus #blurWhenFalse"
-                                                  ng-focus="$ctrl.shouldFocusIfSelectedAnswer()"
-                                                  ng-blur="$ctrl.shouldFocus = false"
-                                                  ng-model="$ctrl.points"
-                                                  ng-keydown="$ctrl.handlePointScroll($event)"
+                            <form (ngSubmit)="savePoints()" class="point-form form-inline">
+                                <label class="inline" *ngIf="showTeacher() || (giveCustomPoints && allowCustomPoints())">
+                                    <ng-container i18n>Points:</ng-container><input class="form-control input-xs"
+                                                  [ngClass]="{ unsaved: points != selectedAnswer?.points,
+                                                  'no-step': !markupSettings.pointsStep }"
+                                                  focus-me="shouldFocus #blurWhenFalse"
```

Comment:
Hmm, huomasin vasta nyt tämän. Ilmeisesti se on 

https://github.com/TIM-JYU/TIM/blob/9f6fe4df733c2a9d5c13d0dc798c09e6e67a5713/timApp/static/scripts/tim/ui/focusMe.ts

tällä ilmeisesti saadaan fokus vaihdettua, kun esim. vaihdetaan AB:n kautta vastaus.

Mutta en nyt huomaa heti eroa tuotannon ja timdevsin välillä, kun asettelen pisteitä ja vaihdan vastauskertoja. Onkohan tuo edes ollenkaan tarpeellinen enää.

Reply:
Näyttäis toimivan ilmankin